### PR TITLE
feat: fetchin salons and single salon; adding salon to favorites; adding opinion to salon

### LINKED ISSUE #9: Delete saloons when deleting bussiness user profile

### PR DESCRIPTION
**Pull request by lzawistowsky**
close #9

### CODE REVIEW

File: controllers/salons.js [hunk 1]
```diff
@@ -2,10 +2,118 @@ const Moment = require('moment');
...
  
 const moment = MomentRange.extendMoment(Moment);
+const mongoose = require('mongoose');
 
 const Salon = require('../models/salon')
 const User = require("../models/user")
 
+exports.getSalons = async (req, res, next) => {
+    try {
+        const city = req.body.city
```
Comment (zawolanski) [inline controllers/salons.js:12]: According to the documentation, there should be `req.body.location`.

File: controllers/salons.js [hunk 2]
```diff
@@ -2,10 +2,118 @@ const Moment = require('moment');
...
+
+        let filter = {}
+        if(city) filter.city = city
+        if(type) filter.type = { $all: type }
+        if(search) filter.name = { $regex: search, $options: "i" }
+
+        const salons = await Salon.find(filter)
+        if(!salons) {
+            const error = new Error('searching salons error')
+            error.statusCode = 500
```
Comment (zawolanski) [inline controllers/salons.js:26]: I think `400` is more appropriate.

**Review by zawolanski — CHANGES_REQUESTED** [review]
Except for a few things, everything is fine.

**Review by pawelsobkowski — COMMENTED** [review]
I think, I have nothing to add

**Pull request merged**